store/memstore/memstore.go

```diff
@@ -34,21 +35,29 @@ func New(maxKeys int) (*MemStore, error) {
 		}
 
 		m = &MemStore{
-			keys: keys,
+			keys:    keys,
+			timeNow: time.Now,
 		}
 	} else {
 		m = &MemStore{
-			m: make(map[string]*int64),
+			m:       make(map[string]*int64),
+			timeNow: time.Now,
 		}
 	}
 	return m, nil
 }
 
+//OverrideTimeNow makes this store use the given function instead of time.Now().
+//This is useful for unit tests that use a simulated wallclock.
+func (ms *MemStore) OverrideTimeNow(timeNow func() time.Time) {
```

Comment:
Would you mind if we called this just `SetTimeNow` since it's basically just a setter for this internal field. This makes it line up nicely with other Go-esque setters like `Request.SetBasicAuth` or `Server.SetKeepAlivesEnabled`.